src/mono/mono/mini/CMakeLists.txt

```diff
@@ -381,6 +381,42 @@ if(NOT DISABLE_SHARED_LIBS)
     add_library(monosgen-shared-dac SHARED "mini-windows-dlldac.c")
     set_target_properties(monosgen-shared-dac PROPERTIES OUTPUT_NAME ${MONO_SHARED_LIB_NAME}-dac)
   endif()
+
+  if(BUILD_DARWIN_FRAMEWORKS)
+    if(TARGET_DARWIN)
```

Comment:
I think this should be `TARGET_DARWIN AND STATIC_COMPONENTS`.
I think this version of the code is only right if the components are static archives, right?

Reply:
So we should never build a .framework in the case of .dylib components? Didn't you say iOS Simulator should ship dylibs & build against stubs in https://github.com/dotnet/runtime/pull/53370#issuecomment-864128856 ?

Reply:
Sorry, I meant to me this looks like the code only supports the static components case.  I assumed a dynamic components version will be added in a follow-up PR. Maybe I misunderstood what's happening here.